src/protocol/tcp/src/main/java/org/apache/jmeter/protocol/tcp/sampler/TCPSampler.java

```diff
@@ -348,8 +350,17 @@ private TCPClient getProtocol() {
         return tcpClient;
     }
 
+    public SampleResult sample(){
+        return sample(null, true);
+    }
+
     @Override
     public SampleResult sample(Entry e)// Entry tends to be ignored ...
+    {
+        return sample(e, firstSample);
+    }
+
+    public SampleResult sample(Entry e, boolean firstSampleFlag)// Entry tends to be ignored ...
```

Comment:
Could you add javadoc explaining firstSampleFlag

Reply:
I have added the java document at this function in new commit.